### ISSUE TITLE
Is use of Blockchain at odds with Right to be forgotten ?

### ISSUE THREAD

**Issue by nelsonic** [dwyl/learn-blockchain#2]
Does the idea of having a **_distributed_ database** where records _cannot_ be deleted without "_breaking_" the chain mean that people _lose_ the "***Right to be forgotten***" ?
See:
+ https://en.wikipedia.org/wiki/Blockchain_(database)
+ https://en.wikipedia.org/wiki/InterPlanetary_File_System
+ https://en.wikipedia.org/wiki/Right_to_be_forgotten
+ https://blog.google/topics/google-europe/reflecting-right-be-forgotten

**Comment by jackie-scholl** [comment]
It depends on what you're using Blockchain for. If you're using it for handling money, [zcash](https://z.cash/) is a highly anonymous public blockchain. If you're using it to store user data, I'd be interested to know what the perceived benefit is over tried and true distributed databases that do allow for editing and deleting records.

**Comment by nelsonic** [comment]
@raptortech-js which distributed database do you have experience with?
(_my interest in Blockchain is for storing sequential records in a system which then allows people to export their data and use it elsewhere..._)

**Comment by jackie-scholl** [comment]
@nelsonic personally, I've used [DynamoDB](https://aws.amazon.com/dynamodb/) and [Amazon Cognito Sync](https://aws.amazon.com/cognito/). For storing purely sequential records I'd use [Amazon S3](https://aws.amazon.com/s3/) (which I also have experience with), which is more of an object store than a database, which makes it (in my mind) even better for this task. If you don't want to use AWS, I believe Azure and Google Cloud Platform both have similar offerings. If you don't want to use the cloud, I'd use [MySQL](https://www.mysql.com/) [with replication](https://dev.mysql.com/doc/refman/5.7/en/replication.html) or possibly [MongoDB](https://www.mongodb.com/). If there was a massive amount of data (50+ TB), I'd consider [Hadoop](https://hadoop.apache.org/). If it was important that it be very highly distributed and fault-tolerant, I'd consider [OrbitDB](https://github.com/haadcode/orbit-db), but be careful; it's still at a very early stage of development.

The major advantages of S3 in this case are that it allows you to drop in data in absolutely any format, that it's very easy to append data, the very attractive cost structure, and the security controls available.

**Comment by nelsonic** [comment]
I tend to agree with Japanese courts on their ruling that _certain_ crimes should not be "forgotten", e.g:
https://www.theguardian.com/world/2017/feb/02/right-to-be-forgotten-online-suffers-setback-after-japan-court-ruling
"_supreme court said, however, that the public’s right to know outweighed the man’s right to privacy, given the serious nature of his crimes_"
